test/unit/compute_resources/ovirt_test.rb

```diff
@@ -24,4 +24,29 @@ class OvirtTest < ActiveSupport::TestCase
       assert_find_by_uuid_raises(ActiveRecord::RecordNotFound, cr)
     end
   end
+
+  describe "associating operating system" do
+    setup do
+      @operating_systems_xml = Nokogiri::XML(File.read('test/fixtures/ovirt_operating_systems.xml'))
+      @compute_resource = FactoryGirl.build(:ovirt_cr).tap do |compute_resource|
+        fog_client = mock
```

Comment:
fog and the ovirt client are irrelevant here if you just stub `available_operating_systems` - what we're testing here is the matching, for which we don't need neither the host (explained in other comments) nor fog, just a valid response from `available_operating_systems`.